xskillscore/core/test.py

```diff
@@ -0,0 +1,110 @@
+import scipy.stats as st
+import xarray as xr
+
+
+def sign_test(
+    forecast1, forecast2, observation, dim=None, categorical=False, alpha=0.05
+):
+    """
+        Returns the Delsole and Tippett sign test over the given time dimension.
+
+        Parameters
+        ----------
+        forecast1 : xarray.Dataset or xarray.DataArray
+            forecast1 to be compared to observation
+        forecast2 : xarray.Dataset or xarray.DataArray
+            forecast2 to be compared to observation
+        observation : xarray.Dataset or xarray.DataArray
+            observation to be compared to both forecasts
+
+            if str, then assume that forecast1 and forecast2 have already been compared
+            to observations and choose from:
+
+                * ``negatively_oriented_already_evaluated``: metric between forecast1
```

Comment:
I agree it's good to allow the ability to use different metrics to determine which forecast is "best". But, instead of the approach you have here, what about allowing the user to specify a `metric` function (and kwargs) that defaults to `xs.mae()`.

Reply:
in the long run it would be great to have metrics with attributes such as `orientation` (from positive, negative).

Reply:
the main reason to introduce this is that for climpred we cannot insert two forecasts into that function but rather return already `xs.metric(forecast,observation)`. now changing string to `xs.metric`, but this is more maintainance heavy that the string method.

Reply:
so we have many entry points to the sign test:
- categorical evaluated inside this function
- categorical evaluated already by climpred/outside of this function
- raw forecast evaluated inside this function
- raw forecast evaluated already by climpred/outside of this function

Reply:
I will implement the metrics approach, if users what to use this test to their own metric, if would be great to have some way to the `sign_test` how to judge.

Reply:
Unfortunately, it's difficult to predict the orientation, especially when we are allowing user-specified functions. I see two appropriate options:

1. Document clearly that `sign_test` is defined such that smaller values are better. It's pretty straightforward for the user to multiply their metric by -1 to account for this.
2. Allow an additional input, `orientation = "positive" / "negative"`.

Reply:
going for 2. orientation='negative' as default.